File: 브루트포스/브론즈/브론즈1/2309_일곱 난쟁이/2309.py
import sys

def find_origin_dwarfs(dwarfs_height):
    all_dwarfs_height = 100
    number_of_dwarfs = len(dwarfs_height)
    total_sum = sum(dwarfs_height)
    for i in range(number_of_dwarfs):
        for j in range(i + 1, number_of_dwarfs):
            if  total_sum - dwarfs_height[i] - dwarfs_height[j] == all_dwarfs_height:
                origin_dwarfs = [dwarfs_height[x] for x in range(number_of_dwarfs) if x != i and x != j]
                return sorted(origin_dwarfs)

#입력값 파싱
input_data = sys.stdin.read()
dwarfs_height = [line for line in map(int, input_data.splitlines())]
origin_dwarf = find_origin_dwarfs(dwarfs_height)
for dwarf in origin_dwarf:
    print(dwarf)

File: 브루트포스/브론즈/브론즈1/2309_일곱 난쟁이/2309.java
import java.util.*;

public class Main {

    public static List<Integer> findOriginDwarfs(List<Integer> dwarfsHeight) {
        int allDwarfsHeight = 100;
        int numberOfDwarfs = dwarfsHeight.size();
        int totalSum = 0;
        
        for (int height : dwarfsHeight) {
            totalSum += height;
        }
        
        for (int i = 0; i < numberOfDwarfs; i++) {
            for (int j = i + 1; j < numberOfDwarfs; j++) {
                if (totalSum - dwarfsHeight.get(i) - dwarfsHeight.get(j) == allDwarfsHeight) {
                    List<Integer> originDwarfs = new ArrayList<>();
                    for (int k = 0; k < numberOfDwarfs; k++) {
                        if (k != i && k != j) {
                            originDwarfs.add(dwarfsHeight.get(k));
                        }
                    }
                    Collections.sort(originDwarfs);
                    return originDwarfs;
                }
            }
        }
        return null; // 찾지 못했을 경우
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        List<Integer> dwarfsHeight = new ArrayList<>();
        
        for (int i = 0; i < 9; i++) {
            dwarfsHeight.add(scanner.nextInt());
        }
        
        List<Integer> originDwarfs = findOriginDwarfs(dwarfsHeight);
        
        for (int dwarf : originDwarfs) {
            System.out.println(dwarf);
        }
        
        scanner.close();
    }
}
